pkg/app/api/grpcapi/web_api.go

```diff
@@ -184,6 +185,8 @@ func (a *WebAPI) DisableEnvironment(ctx context.Context, req *webservice.Disable
 	return &webservice.DisableEnvironmentResponse{}, nil
 }
 
+// DeleteEnvironment deletes the given environment and all applications that belongs to it.
+// It returns a FailedPrecondition error if Any Piped has permission to that environment.
```

Comment:
nit: if any Piped is referencing to that environment (or if any Piped is still using that environment.)

Reply:
That's obviously easier to understand for users. (I sometimes get caught up with the developer's point of view)

Reply:
👍